2023-06-25 12:27:26,426 [154] FATAL WasteManagement.Controllers.InvoiceController [(null)] - log4net Fatal Level
System.Data.Entity.Validation.DbEntityValidationException: Validation failed for one or more entities. See 'EntityValidationErrors' property for more details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at WasteManagement.Controllers.InvoiceController.Index(InvocieViewModel invoice, String Create, String Update, String Multiple, HttpPostedFileBase AttachedDocuments, HttpPostedFileBase AttachedDocumentUpdate) in D:\WasteManagement\WasteManagement\Controllers\InvoiceController.cs:line 288
2023-06-25 12:30:09,026 [157] FATAL WasteManagement.Controllers.InvoiceController [(null)] - log4net Fatal Level
System.Data.Entity.Validation.DbEntityValidationException: Validation failed for one or more entities. See 'EntityValidationErrors' property for more details.
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at WasteManagement.Controllers.InvoiceController.Index(InvocieViewModel invoice, String Create, String Update, String Multiple, HttpPostedFileBase AttachedDocuments, HttpPostedFileBase AttachedDocumentUpdate) in D:\WasteManagement\WasteManagement\Controllers\InvoiceController.cs:line 288
2023-06-25 13:22:05,648 [138] FATAL WasteManagement.Controllers.InvoiceController [(null)] - log4net Fatal Level
System.NullReferenceException: Object reference not set to an instance of an object.
   at WasteManagement.Controllers.InvoiceController.Index(InvocieViewModel invoice, String Create, String Update, String Multiple, HttpPostedFileBase AttachedDocuments, HttpPostedFileBase AttachedDocumentUpdate) in D:\WasteManagement\WasteManagement\Controllers\InvoiceController.cs:line 399
2023-06-25 13:33:33,215 [19] FATAL WasteManagement.DataLayer.WasteManagementTransactions [(null)] - Check Transaction: Inv-2562023-003: Opening: 0 :Closing: 696
2023-07-05 21:54:32,002 [192] FATAL WasteManagement.DataLayer.WasteManagementDBAccess [(null)] - log4net Fatal Level
System.NullReferenceException: Object reference not set to an instance of an object.
   at WasteManagement.DataLayer.WasteManagementDBAccess.GetLastLedgerCode() in D:\WasteManagement\WasteManagement\DataLayer\MoneyExchangerDBAccess.cs:line 2642
2023-07-05 21:55:04,284 [195] FATAL WasteManagement.DataLayer.WasteManagementDBAccess [(null)] - log4net Fatal Level
System.NullReferenceException: Object reference not set to an instance of an object.
   at WasteManagement.DataLayer.WasteManagementDBAccess.GetLastBankNumber() in D:\WasteManagement\WasteManagement\DataLayer\MoneyExchangerDBAccess.cs:line 2052
2023-07-05 21:57:32,754 [163] FATAL WasteManagement.Controllers.InvoiceController [(null)] - log4net Fatal Level
System.Data.Entity.Infrastructure.DbUpdateException: An error occurred while updating the entries. See the inner exception for details. ---> System.Data.Entity.Core.UpdateException: An error occurred while updating the entries. See the inner exception for details. ---> System.Data.SqlClient.SqlException: The conversion of a datetime2 data type to a datetime data type resulted in an out-of-range value.
The statement has been terminated.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at System.Data.SqlClient.SqlDataReader.get_MetaData()
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry)
   at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at System.Data.SqlClient.SqlCommand.ExecuteReader(CommandBehavior behavior, String method)
   at System.Data.Entity.Infrastructure.Interception.InternalDispatcher`1.Dispatch[TTarget,TInterceptionContext,TResult](TTarget target, Func`3 operation, TInterceptionContext interceptionContext, Action`3 executing, Action`3 executed)
   at System.Data.Entity.Infrastructure.Interception.DbCommandDispatcher.Reader(DbCommand command, DbCommandInterceptionContext interceptionContext)
   at System.Data.Entity.Core.Mapping.Update.Internal.DynamicUpdateCommand.Execute(Dictionary`2 identifierValues, List`1 generatedValues)
   at System.Data.Entity.Core.Mapping.Update.Internal.UpdateTranslator.Update()
   --- End of inner exception stack trace ---
   at System.Data.Entity.Core.Mapping.Update.Internal.UpdateTranslator.Update()
   at System.Data.Entity.Core.Objects.ObjectContext.ExecuteInTransaction[T](Func`1 func, IDbExecutionStrategy executionStrategy, Boolean startLocalTransaction, Boolean releaseConnectionOnSuccess)
   at System.Data.Entity.Core.Objects.ObjectContext.SaveChangesToStore(SaveOptions options, IDbExecutionStrategy executionStrategy, Boolean startLocalTransaction)
   at System.Data.Entity.SqlServer.DefaultSqlExecutionStrategy.Execute[TResult](Func`1 operation)
   at System.Data.Entity.Core.Objects.ObjectContext.SaveChangesInternal(SaveOptions options, Boolean executeInExistingTransaction)
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   --- End of inner exception stack trace ---
   at System.Data.Entity.Internal.InternalContext.SaveChanges()
   at WasteManagement.Controllers.InvoiceController.PaymentReceived(AllReceivedPayments allpayment, String Create, String Update, String Multiple, HttpPostedFileBase AttachedDocuments, HttpPostedFileBase AttachedDocumentUpdate) in D:\WasteManagement\WasteManagement\Controllers\InvoiceController.cs:line 1271
